### PR TITLE
Add listen_ip, listen_port, hidden and advanced parameters. Add fallback parser.

### PR DESCRIPTION
**Pull request by ericwholt**
This update is to simplify setup by separating ip and port into different fields. We are maintaining `listen_address` for backwards compatability. Also added parameters unused by stanza. These will be used by a UI to hide parameters based on `advanced` and `hidden` parameters 

  - Added `listen_ip` as parameter
  - Added `listen_port` as parameter
  - Added parameter `advanced_config` set to true on `listen_ip`. This is to be used in UI to hide `listen_ip` behind an advanced option
  - Added parameter `listen_address`. The parameter `listen_address` is being maintained for backwards compatibility.
  - Added fall back regex parser incase syslog parser fails. This will attempt to parse severity from priority. The unparsed portion will be put in message.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,10 +4,46 @@ All notable changes to this project will be documented in this file.
...
+- Update `cisco_meraki` plugin ([PR316](https://github.com/observIQ/stanza-plugins/pull/316))
+  - Added `listen_ip` as parameter
+  - Added `listen_port` as parameter
+  - Changed description of `listen_address` to "Parameter Deprecated Use `listen_ip` and `listen_port` instead."
+- Update `syslog`, `rsyslog`, `syslogng` plugin ([PR316](https://github.com/observIQ/stanza-plugins/pull/316))
+  - Added `listen_ip` as parameter
+  - Added `listen_port` as parameter
+  - Added parameter `advanced_config` set to true on `listen_ip` and `location`. This is to be used in UI to hide `listen_ip` behind an advanced option
+  - Changed description of `listen_address` to "Parameter Deprecated Use `listen_ip` and `listen_port` instead."
+  - Added fall back regex parser incase syslog parser fails. This will attempt to parse severity from priority. The unparsed portion will be put in message.
```
Comment (jsirianni) [inline CHANGELOG.md:41]: Can the size of this be reduced? Perhaps group all the identical changes?

**Review by jsirianni — CHANGES_REQUESTED** [review]
Please add test cases for the new parameters for all changed plugins.

File: plugins/cisco_meraki.yaml [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
+  - name: listen_ip
+    label: Listen Address
+    description: A syslog ip address of the form `<ip>`
+    type: string
+    default: "0.0.0.0"
+    advanced_config: true
   - name: listen_address
     label: Listen Address
-    description: A syslog address of the form `<ip>:<port>`
+    description: Parameter Deprecated Use `listen_ip` and `listen_port` instead.
```
[the remark this answers is not in the document]
Comment (ericwholt) [reply plugins/cisco_meraki.yaml:19]: It was intentional. I had originally intended to hide the old listen address. I will update the new listen address to Listen IP instead.

File: CHANGELOG.md (hunk 1 above)
Comment (ericwholt) [reply CHANGELOG.md:41]: Reduced all syslog entries to one list.

**Comment by jsirianni** [comment]
I get this error while running the agent against this repo 
`./artifacts/stanza_darwin_amd64 --plugin_dir ~/git/stanza-plugins/plugins --config ./config_cisco_asa.yaml`
```json
{
  "level": "error",
  "timestamp": "2021-09-08T14:18:07.992-0400",
  "message": "Got errors parsing plugins",
  "errors": [
    {
      "error": "parse plugin file: could not unmarshal plugin file: template: Syslog-ng:13: undefined variable \"$length\": {\"path\":\"/Users/jsirianni/git/stanza-plugins/plugins/syslogng.yaml\"}"
    }
  ]
}
```

File: plugins/rsyslog.yaml [hunk 3]
```diff
@@ -68,4 +83,21 @@ pipeline:
...
+      parse_from: $record.priority
+      mapping:
+        emergency: [0,8,16,24,32,40,48,56,64,72,80,88,96,104,112,120,128,136,144,152,160,168,176,184]
+        alert: [1,9,17,25,33,41,49,57,65,73,81,89,97,105,113,121,129,137,145,153,161,169,177,185]
+        critical: [2,10,18,26,34,42,50,58,66,74,82,90,98,106,114,122,130,138,146,154,162,170,178,186]
+        error: [3,11,19,27,35,43,51,59,67,75,83,91,99,107,115,123,131,139,147,155,163,171,179,187]
+        warning: [4,12,20,28,36,44,52,60,68,76,84,92,100,108,116,124,132,140,148,156,164,172,180,188]
+        notice: [5,13,21,29,37,45,53,61,69,77,85,93,101,109,117,125,133,141,149,157,165,173,181,189]
+        info: [6,14,22,30,38,46,54,62,70,78,86,94,102,110,118,126,134,142,150,158,166,174,182,190]
+        debug: [7,15,23,31,39,47,55,63,71,79,87,95,103,111,119,127,135,143,151,159,167,175,183,191]
```
[the remark this answers is not in the document]
Comment (jsirianni) [reply plugins/rsyslog.yaml:103]: Expanding on this, lets remove it from syslog-ng and rsyslog. We should be able to trust these formats to adhere to the standards.

[the remark this answers is not in the document]
Comment (ericwholt) [reply plugins/rsyslog.yaml]: I removed the fallback changes in `rsyslog` and `syslogng`.

**Comment by ericwholt** [comment]
I have add some additional tests for a few of the syslog plugins. I will circle back to this in near future.

**Review by jsirianni — APPROVED** [review]

**Pull request merged**